src/lib.rs

```diff
@@ -589,6 +484,76 @@ pub struct KeystoreSession<'a, L: Ledger, Backend: KeystoreBackend<'a, L>> {
 }
 
+impl<'a, L: Ledger, Backend: KeystoreBackend<'a, L>, Meta: Serialize + DeserializeOwned + Send>
+    KeystoreSession<'a, L, Backend, Meta>
+{
+    /// The storage layer used by this implementation.
+    // type Storage: KeystoreStorage<'a, L> + Send;
+
+    /// Load a keystore or create a new one.
+    ///
+    /// If a keystore exists in storage, it is loaded. Otherwise, [KeystoreBackend::create] is called to
+    /// create a new keystore.
+    // async fn load(&mut self) -> Result<KeystoreState<'a, L>, KeystoreError<L>> {
+    //     let mut storage = self.storage.lock().await;
+    //     if storage.exists() {
+    //         // If there is a stored keystore with this key pair, load it.
+    //         storage.load().await
+    //     } else {
+    //         // Otherwise, ask the network layer to create and register a brand new keystore.
+    //         let state = self.backend.create().await?;
+    //         storage.create(&state).await?;
+    //         Ok(state)
+    //     }
+    // }
+
+    /// Make a change to the persisted state using a function describing a transaction.
+    ///
+    /// # Example
+    ///
+    /// ```ignore
+    /// backend.store(key_pair, |mut t| async move {
+    ///     t.store_snapshot(keystore_state).await?;
+    ///     // If this store fails, the effects of the previous store will be reverted.
+    ///     t.store_asset(keystore_state, asset).await?;
+    ///     // Use `t.backend` to access other backend functions during the transaction. Any
+    ///     // failures here will revert all previous stores.
+    ///     t.backend.do_something().await?;
+    ///     Ok(t)
+    /// }).await?;
+    /// ```
+    async fn store<'l, F, Fut>(&'l mut self, update: F) -> Result<(), KeystoreError<L>>
+    where
+        F: Send + FnOnce(StorageTransaction<'a, 'l, L, Backend, Meta>) -> Fut,
+        Fut: Send
+            + Future<Output = Result<StorageTransaction<'a, 'l, L, Backend, Meta>, KeystoreError<L>>>,
+    {
+        let fut = update(StorageTransaction::new(
+            &mut self.backend,
+            self.storage.clone(),
+        ))
+        .and_then(|mut txn| async move {
+            txn.storage().await.commit().await;
+            Ok(())
+        });
+        fut.await
+    }
+
+    /// Access the persistent storage layer.
+    ///
+    /// Note that the return type of this function requires the implementation to guard the storage
+    /// layer with a mutex, even if it is not internally shared between threads. This is meant to
+    /// allow shared access to the storage layer internally, but not to require it. A better
+    /// interface would be to have an associated type
+    ///         `type<'l> StorageRef: 'l +  Deref<Target = Self::Storage> + DerefMut`
+    /// This could be [MutexGuard], [async_std::sync::RwLockWriteGuard], or just
+    /// `&mut Self::Storage`, depending on the needs of the implementation. This should be cleaned
+    /// up if and when GATs stabilize.
```

Comment:
I think this comment is irrelevant, now that there is only a single implementation and no associated type